tests/test-buttercup.el

```diff
@@ -1107,11 +1109,32 @@
     (spy-on 'send-string-to-terminal))
 
   (it "should send a formatted string to the terminal"
-    (buttercup--print "Hello, %s" "world")
+    (let ((buttercup--pending-output t))
+      (buttercup--print "Hello, %s" "world"))
 
     (expect 'send-string-to-terminal
             :to-have-been-called-with
-            "Hello, world")))
+            "Hello, world"))
+
+  (it "should put output on hold"
+    (let ((buttercup--pending-output nil))
+      (buttercup--print "not to be printed just yet")
+
+      (expect 'send-string-to-terminal :to-have-been-called-times 0)
+      (expect buttercup--pending-output :to-equal '("not to be printed just yet"))))
+
+  (it "should discard one level of output"
+    (let ((buttercup--pending-output nil))
+      (buttercup--print "1")
+      (buttercup--start-output-level)
+      (buttercup--print "2")
+      (buttercup--discard-one-output-level)
+      (buttercup--print "3")
+      (buttercup--flush-pending-output))
```

Comment:
Add an expect eher to check that send-string-to-terminal has not been called yet.

Reply:
Done.